nixos/modules/services/misc/osticket/default.nix

```diff
@@ -0,0 +1,282 @@
+        '';
+      };
+
+      # TODO make this nicer wrt getting the plugins, perhaps ship all plugins by default + allow extra
+      plugins = mkOption {
+        type = with types; attrsOf path;
+        default = { };
+        description = ''
+          attrset with paths to plugins, like in nixos/modules/services/misc/osticket/plugins.nix
+        '';
+      };
+
+      enablePolling = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Whether to run email polling.";
+      };
+
+      poolConfig = mkOption {
+        type = with types; attrsOf (oneOf [ str int bool ]);
+        default = {
+          "pm" = "dynamic";
+          "pm.max_children" = 32;
+          "pm.start_servers" = 1;
+          "pm.min_spare_servers" = 1;
+          "pm.max_spare_servers" = 2;
+          "pm.max_requests" = 500;
+          "php_value[date.timezone]" = "UTC";
+          "php_value[upload_max_filesize]" = "10M";
+          # Redirect worker stdout and stderr into main error log. If not set, stdout and
+          # stderr will be redirected to /dev/null according to FastCGI specs.
+          # Default Value: no
+          "catch_workers_output" = true;
+          "clear_env" = false;
+          "env[\"SHELL_VERBOSITY\"]" = 3;
+        };
+        description = ''
+          Options for the osTicket PHP pool. See the documentation on <literal>php-fpm.conf</literal>
+          for details on configuration directives.
+        '';
+      };
+
+      virtualHost = mkOption {
+        type = types.submodule (import ../../web-servers/nginx/vhost-options.nix);
+        example = literalExpression ''
+          {
+            serverName = "osticket.example.org";
+            forceSSL = true;
+            enableACME = true;
+          }
+        '';
+        description = ''
+          Nginx configuration can be done by adapting <option>services.nginx.virtualHosts</option>.
+          See <xref linkend="opt-services.nginx.virtualHosts"/> for further information.
+        '';
+      };
+
+    };
+  };
+
+  # implementation
+  config = mkIf cfg.enable (
+    let
+      group = config.services.nginx.group;
+      fpm = config.services.phpfpm.pools.${user};
+
+      myPhp = pkgs.php74.buildEnv {
+        extensions = { enabled, all }:
+          enabled ++ (with all; [
+            apcu
+            gd
+            gettext
+            imap
+            json
+            mysqli
+            mbstring
+            # required but already in enabled
+            # xml
+            # Needed for Slack plugin
+            # TODO extensions function on plugin definition
+            #   that then gets mapped and joined
+            curl
+          ]);
+        # extraConfig = "memory_limit=256M";
+      };
+
+      # The deploy uses git to get the version
+      fakeGit = pkgs.writeScriptBin "git" ''
+        #!${pkgs.stdenv.shell}
+        echo ${version}
+      '';
+
+      pkg = pkgs.stdenv.mkDerivation rec {
```

Comment:
Is there a reason you're not adding this derivation to the package set and instead putting it directly in the module?

Reply:
the derivation needs the database user etc. Without it, it won't work, and it doesn't make much sense as a stand-alone package.

That's why I thought it would be better together in the module.

Reply:
In fact `mediawiki` **does** have a [package](https://github.com/NixOS/nixpkgs/blob/0699530f08290f34c532beedd66046825d9756fa/pkgs/servers/web-apps/mediawiki/default.nix). What you linked to was a bit of a wrapper around the package. You should **definitely** create a separate derivation file in the `pkgs/` tree... how else will our bots automatically update your software?